runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -1256,11 +1257,37 @@ struct fi_info* findProvInList(struct fi_info* info,
                                chpl_bool accept_ungood_provs,
                                chpl_bool accept_RxD_provs,
                                chpl_bool accept_RxM_provs) {
-  while (info != NULL
-         && (   (!accept_ungood_provs && !isGoodCoreProvider(info))
-             || (!accept_RxD_provs && isInProvider("ofi_rxd", info))
-             || (!accept_RxM_provs && isInProvider("ofi_rxm", info)))) {
-    info = info->next;
+
+  // Ignore any provider for the touchbar interface on a mac. As of v1.13.0 libfabric does
+  // not do this internally. All touchbars currently have the same MAC address. 
+  struct sockaddr_in6 touchbar;
+  chpl_bool darwin = !strcmp(CHPL_TARGET_PLATFORM, "darwin");
+  if (darwin) {
+    int rc = inet_pton(AF_INET6, "fe80::aede:48ff:fe00:1122", &touchbar.sin6_addr);
```

Comment:
Putting this explicit static address in the code makes me nervous, because we don't know why it is what it is and thus, whether or not it might change and break our code. So I poked around in the interwebs and it looks like this is actually not the touchbar as such, but more generally the internal network interface to the so-called "T2 coprocessor". (See [here](https://duo.com/labs/research/apple-t2-xpc).) That being so, is there some way to determine this address at execution time by looking at the system configuration? As a start, does this show up in the output of `ifconfig`?

Reply:
Yes, I guess technically it's the T2 coprocessor. I'll change the comments. I don't know of a good way to identify it other than its fixed local address of fe80::aede:48ff:fe00:1122. It does show up in ifconfig but I'm not sure how that helps me. I'm hoping all of this is a temporary fix until libfabric filters out this interface internally.

Reply:
Would you check on the libfabric-users mailing list and see if we can get any advice there, either about avoiding it or when (if?) they plan on adjusting provider(s) not to show it?

Reply:
I did and I got no response.

Reply:
Oh, this is weird. I couldn't find the result of that post in my email, including even in my deleted folder. But now that you mention it I do recall seeing it. And I can find it in the online archives of the mailing list ([here](https://lists.openfabrics.org/pipermail/libfabric-users/2021-August/000869.html)).

Okay, if there aren't any differences between the provider that corresponds to this internal bridge and the actual external network interface then I guess we'll just have to match it explicitly.

I'll do a review acceptance separately.